unittests/runtime/Mutex.cpp

```diff
@@ -13,12 +13,57 @@
 #include "swift/Runtime/Mutex.h"
 #include "gtest/gtest.h"
 #include <chrono>
+#include <map>
+#include <random>
 #include <thread>
 
 using namespace swift;
 
+// When true many of the threaded tests log activity to help triage issues.
+static bool trace = false;
+
+// Note the use of non-atomic booleans for "lazy" thread control in many of
+// the tests is sufficient for what we are doing here, no need for
+// atomic operations.
+
+template <typename ThreadBody, typename AfterSpinRelease>
+void threadedExecute(int threadCount, ThreadBody threadBody,
+                     AfterSpinRelease afterSpinRelease) {
+
+  std::vector<std::thread> threads;
+
+  // Block the threads we are about to create.
+  bool spinWait = true;
```

Comment:
> It sounds exactly like a description of a read/write race to me.

Except that the readers are looping so they may get the wrong old value at one put but then will get the correct value in the near future (sufficient for the usage pattern in this code), assuming the loads aren't optimized away, note barriers are happening in this code given what it is doing, I am running under that assumption in my use of lazy booleans. It could still fall in undefined territory despite my current understanding of the liberties of the optimizer in this particular code structure.

I have changed it to use std::atomic to avoid the potential pitfalls. Thanks for the information.